### PR TITLE
Runtime compressed refs work

### PR DESCRIPTION
**Pull request by gacholio**
Get the P JIT working in mixed mode. This PR does not address the crypto
code.

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
@gita-omr @zl-wang Can you please find someone to review this?

**Comment by zl-wang** [comment]
i will try to do it tomorrow

File: runtime/compiler/p/runtime/PicBuilder.spp [hunk 1]
```diff
@@ -64,6 +64,22 @@
...
 
+#if defined(OMR_GC_COMPRESSED_POINTERS)
+#if defined(OMR_GC_FULL_POINTERS)
+#define LOAD_CLASS(dest, src) \
+	laddr dest,J9TR_VMThreadCompressObjectReferences(r15) ; \
+	cmpli cr0, CmpAddr, dest, 0 ; \
+	beq $+12 ; \
+	lwz dest, J9TR_J9Object_class(src) ; \
+	b $+8 ; \
+	laddr dest, J9TR_J9Object_class(src)
```
Comment (zl-wang) [inline runtime/compiler/p/runtime/PicBuilder.spp]: rather having a taken branch always, we can have a sequence of no taken branch for one way:

laddr  dest, J9TR_VMThreadCompressObjectReferences(r15) 
cmpli cr0, CmpAddr, dest, 0 
laddr dest, J9TR_J9Object_class(src)
beq  +8
lwz dest, J9TR_J9Object_class(src)

**Comment by zl-wang** [comment]
a quick scan on the crypto assembly didn't show they are loading/storing clazz field.  likely no change is needed.

**Comment by gacholio** [comment]
> a quick scan on the crypto assembly didn't show they are loading/storing clazz field. likely no change is needed.

The crypto code references J9IndexableObjectContiguous_objectData, which now needs to be split for compressed and full. See the X86 version for what I did before - I just duplicated the entry points for compressed and full, and backended on the common code.

File: runtime/compiler/p/runtime/PicBuilder.spp (hunk 1 above)
Comment (gacholio) [reply runtime/compiler/p/runtime/PicBuilder.spp]: Fixed.

Comment (gacholio) [reply runtime/compiler/p/runtime/PicBuilder.spp]: Verified that mixed mode still works after this change.

**Comment by zl-wang** [comment]
Jenkins test extended plinux,aix jdk8,jdk11

**Pull request merged**